solana/src/main/java/com/solana/networking/RpcResponse.kt

```diff
@@ -16,10 +16,29 @@ typealias DefaultRpcResponse = RpcResponseSerializable<JsonElement>
 data class RpcError(val code: Int, val message: String)
 
 @Serializable
-open class RpcResponseSerializable<R>(
-    open val result: R? = null,
-    open val error: RpcError? = null,
-    val id: String? = null
+data class RpcResponseSerializable<R>(
+    val result: R? = null,
+    val error: RpcError? = null,
+    val id: String? = null,
+
+    // Sockets
+    val params: Params<R>? = null
```

Comment:
is there a way we can make a dedicated response class for sockets? 

kotlin.serialization has some tools that might help with this. something like this maybe?
```
data class SocketResponse<R>(
    @SerialName("params") override val result: Params<R>?,
    ...
) : RpcResponseSerializable
```

Reply:
I tried the change. We can not override result since result can be used or even returned on some request. Normally is used to confirm the socketId of the subscription.

Reply:
We can add another object that copies the same as RpcResponse or leave it as is with the Params. We can also inherit the data class or implement an interface. I want to hear your opinon

Reply:
> We can not override result since result can be used or even returned on some request.
The code I added above was just an example. My point was that we should create anew object that describes whatever the json data structure should be for this socket response in order to keep in line with SOLID principles. 

I think we should make a base class or interface that defines the base response and subclass the socket response. 

Sealed class might be appropriate here, or the kotlin decoders can be made to select a serializer class based on the json content

Reply:
I went to the interface route because we can have a metaplex response with the new RPCs for read api